vignettes/value-labels.Rmd

```diff
@@ -117,134 +156,173 @@ HEALTH2 <- ifelse(cps$HEALTH > 3, 3, cps$HEALTH)
-cps$STATEFIP <- as_factor(cps$STATEFIP)
+typeof(inctot_num)
+
+ipums_val_labels(inctot_num)
 ```
 
-## `lbl_na_if()`
-The INCTOT variable has 2 labelled values that are not actually incomes:
-99999998 indicates "Missing" and 99999999 indicate "Not in Universe".
-On the CPS website, the Universe tab indicates that the Universe for 2016
-is respondents age 15+. Let's say for my analysis, I can treat these values
-as missing. 
-
-The `lbl_na_if()` function takes the variable and a function that refers to
-.val and .lbl (the values and labels respectively) and returns an indicator
-of whether to set those values to NA and remove the label. You can also use the 
-`~` notation from the purrr package to create succinct anonymous functions.
-
-The .val and .lbl only refer to values that already have labels, they do not apply 
-to unlabeled values. See the `lbl_add()` and `lbl_add_vals()` functions below for
-working with unlabeled values.
+Note that labelled values are not generally intended to be interpreted
+as numeric values, so `zap_labels()` should only be used after labels
+have been properly handled. For example, in `INCTOT`, labelled values
+used to identify missing values are encoded with large numbers:
+
 ```{r}
-# Caution: R defaults to printing large numbers like 99999999 in rounded 
-# exponential format (1e+08) but that's not how they are actually stored
 ipums_val_labels(cps$INCTOT)
+```
+
+Treating these as legitimate observations will significantly skew any
+calculations with this variable if not first converted to `NA`.
+
+### Create missing values based on value labels
 
-# All of these are equivalent
-INCTOT1 <- lbl_na_if(cps$INCTOT, ~.val >= 99999990)
-INCTOT2 <- lbl_na_if(cps$INCTOT, ~.lbl %in% c("Missing.", "N.I.U. (Not in Universe)."))
-INCTOT3 <- lbl_na_if(cps$INCTOT, function(.val, .lbl) {
-  is_missing <- .val == 99999998
-  is_niu <- .lbl == "N.I.U. (Not in Universe)."
-  return(is_missing | is_niu)
-})
-
-# Change to a factor in the original cps data.frame
-cps$INCTOT <- lbl_na_if(cps$INCTOT, ~.val >= 9999990)
-cps$INCTOT <- as_factor(cps$INCTOT)
+Many IPUMS variables use labelled values to identify missing data. This
+allows for more detail about why certain observations were missing than
+would be available were values loaded as `NA`.
+
+As we saw with `INCTOT`, value labels were used to identify two types of
+missing data: those that are legitimately missing and those that are not
+in the universe of observations.
+
+```{r}
+ipums_val_labels(cps$INCTOT)
 ```
 
-## `lbl_collapse()`
-The EDUC variable provides an example of a common IPUMS practice of grouping
-categories together by the starting digits. For example, the value 10 indicates 
-"Grades 1, 2, 3, or 4", and 11 - "Grade 1", 12 - "Grade 2", etc. 
+To convert one or both of these labelled values to `NA`, use
+`lbl_na_if()`. `lbl_na_if()` uses a function of the value labels to
+handle the conversion. The function should take each value label as its
```

Comment:
Here I think you are using the term "value label" to refer to what I would call a "value and value label pair". When I say "value label", I am referring only to the label, not the value and associated label. I think this is how others on the microdata side use these terms, and I think it squares with general usage of those terms. For example, an excerpt from the introductory vignette for the labelled package (emphasis added):

> Use val_labels() to get all value labels and val_label() to get the *value label associated with a specific value*.

I changed this particular instance to "value and value label pair", but we might need more discussion to get on the same page with this. I've recently learned that IT uses the term "category" to encompass a value and value label pair. We could introduce that concept here, but I think there's a possibility that could actually make things more confusing for some users.